src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsSolutionRestoreService4.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Represents a package restore service API for integration with a project system.
+    /// Implemented by NuGet.
```

Comment:
I'm confused. If it's implemented by us, where is it?

Reply:
This is just the API so that the project-system can start working on their end. 

We'll only implement it on NuGet side later when the project system has implemented their side, because we need to perform a lot of tests on our end.

The implementation will be handled in https://github.com/NuGet/Home/issues/10678. 

I realize I didn't put that detail in the PR description but an issue :D .